pkg/reconciler/testing/messagingchannel.go

```diff
@@ -111,6 +112,35 @@ func WithMessagingChannelSubscribers(subscribers []eventingduckv1alpha1.Subscrib
 	}
 }
 
+func WithMessagingChannelReadySubscriber(uid string) MessagingChannelOption {
```

Comment:
Is this one needed? Or does everything call WithMessagingChannelReadySubscriberAndGeneration?

Reply:
For some test cases, it needs WithMessagingChannelReadySubscriber. It's because some unit test didn't set generation for subscription. I think I can make WithMessagingChannelReadySubscriber() to call WithMessagingChannelReadySubscriberAndGeneration with generation parameter 0 to get rid of the duplicate.